stitching/__main__.py

```diff
@@ -1,115 +1,354 @@
+		self.canvas3.place(x=20,y=210)
+		
+		self.label8 = Label(self.canvas3, text="Select cell", fg='blue')
+		self.label8.configure(width = 13)
+		self.label8.place(x=70,y=40)
+		
+		self.label9 = Label(self.canvas3, text="Select channel", fg='blue')
+		self.label9.configure(width = 13)
+		self.label9.place(x=70,y=120)
+		
+		self.entry4 = Entry(self.canvas3,width = 13)
+		self.entry4.place(x = 70,y=58)
+		
+		self.entry5 = Entry(self.canvas3,width= 13)
+		self.entry5.place(x = 70,y=138) 
+		
+		self.button3 = Button(self.canvas3, text = "Display image", command = self.display_cell)
+		self.button3.configure(width = 20, height = 3, background = "#33B5E5")
+		self.button3.place(x=30,y=180)
+		
+		
+	def messageBox1(self):
+		text = "Select channel numbers. Separate each channel or range with a comma (such as 1,3,5-7). Per default all channels will be used."
+		tkMessageBox.showinfo("", text)
+		
+	def messageBox2(self):
+		text = "Enter the number of cells per row and per column in the output image. The default value for the grid size is 30 and will yield output images with 30x30 cells per montage. The maximum possible value is 100."
+		tkMessageBox.showinfo("", text)
+		
+	def chooseDir(self):
+		self.dirPath = tkFileDialog.askdirectory()
+		self.Text4.config(state=NORMAL)
+		self.Text4.delete(1.0, END)
+		self.Text4.insert(END,self.dirPath)
+		self.Text4.config(state=DISABLED)
+		
+	def read_cif(self):
+		old_file = self.filename
+		self.filename = tkFileDialog.askopenfilename()
+		if type(self.filename) != tuple:
+			if self.filename.lower().endswith('.cif'):
+				try:
+					self.Text1.config(state=NORMAL)
+					self.Text2.config(state=NORMAL)
+					self.Text3.config(state=NORMAL)
+					self.Text1.delete(1.0, END)
+					self.Text2.delete(1.0, END)
+					self.Text3.delete(1.0, END)
+					self.entry1.delete(0,END)
+					self.entry2.delete(0,END)
+			
+					javabridge.start_vm(class_path=bioformats.JARS, max_heap_size='8G')
+					self.reader = bioformats.formatreader.get_image_reader("tmp", path=self.filename)
+					self.n_images = int(0.5*javabridge.call(self.reader.metadata, "getImageCount", "()I"))
+					self.n_channels = javabridge.call(self.reader.metadata, "getChannelCount", "(I)I", 0)
+
+					#check, whether the .cif file is ok
+					if self.n_images == 0 or self.n_channels == 0:
+						raise ValueError('There is nothing in the file')
+					self.Text1.insert(END,self.filename)
+					self.Text1.config(state=DISABLED)
+					self.Text2.insert(END,self.n_images)
+					self.Text2.config(state=DISABLED)
+					self.Text3.insert(END,self.n_channels)
+					self.Text3.config(state=DISABLED)
+					self.entry1.insert(0,'1-'+str(self.n_channels))
+					self.entry2.insert(0,30)
+				except:
+					tkMessageBox.showinfo("Error", "Please choose a correct .cif file.")
+			if self.filename and not self.filename.lower().endswith('.cif'):
+				tkMessageBox.showinfo("Error", "Please choose a .cif file.")
+			
+	def generate_tifs(self):
+		if self.reader != 0:
```

Comment:
Please refactor this function to reduce the number of nested if/else and try/except blocks. For example,

```python
if self.reader == 0:
    tkMessageBox.showinfo("Error", "Please choose a .cif file.")
    return

if type(self.dirPath) == tuple or not os.path.isdir(self.dirPath):
    tkMessageBox.showinfo("Error", "Please choose an output directory.")
    return

...
```